src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -166,29 +166,20 @@ private async Task ProcessQueueItems(object[] queueItems)
 
 				if (result != null && !result.IsSuccessStatusCode)
 				{
-					var str = await result.Content.ReadAsStringAsync();
-					_logger.LogError("Failed sending event. {ApmServerResponse}", str);
+					_logger?.IfLevel(LogLevel.Error)?.Log("Failed sending event. {ApmServerResponse}", await result.Content.ReadAsStringAsync());
```

Comment:
Small suggestion what if `IfLevel()` is private and we expose the different levels as methods:

```suggestion
					_logger?.Debug()?.Log("Failed sending event. {ApmServerResponse}", await result.Content.ReadAsStringAsync());
```

IMHO we can then remove the other extensions methods on `IApmLogger` such as `LogWarning()` et all.

Reply:
👍  Done, nice idea!